src/cachelib/base.py

```diff
@@ -1,20 +1,84 @@
+import base64
+import io
+import pickle
+import typing as _t
+
+import itsdangerous
+
+
+class _Base85Pickler:
+    """
+    Pickles Base85 safe. To allow multiple reads from the same file, encoded output must
+    not contain any newlines. Base85 has less overhead than Base64.
+    """
+
+    @staticmethod
+    def dumps(obj: _t.Any) -> bytes:
+        return base64.b85encode(pickle.dumps(obj))
+
+    @staticmethod
+    def loads(data: _t.AnyStr) -> bytes:
+        return pickle.loads(base64.b85decode(data))
+
+
 class BaseCache:
     """Baseclass for the cache systems.  All the cache systems implement this
     API or a superset of it.
 
     :param default_timeout: the default timeout (in seconds) that is used if
                             no timeout is specified on :meth:`set`. A timeout
                             of 0 indicates that the cache never expires.
+
+    :param secret_key: Key to sign cache entries with.
+
+        .. versionadded:: 0.3.0
+
     """
 
-    def __init__(self, default_timeout=300):
+    def __init__(
+        self,
+        default_timeout=300,
+        *,
+        secret_key: _t.Optional[_t.Union[_t.AnyStr, _t.Collection[_t.AnyStr]]] = None
+    ):
         self.default_timeout = default_timeout
+        if secret_key is not None:
+            self.__signed_serializer = itsdangerous.Serializer(
```

Comment:
Why double-underscore names? That makes subclassing less convenient.

Reply:
Subclass are only supposed to access `_dump/load(s)`, but sure, I can remove a single underscore.